.github/workflows/tests.yml

```diff
@@ -16,114 +16,44 @@ jobs:
+          - os: windows-latest
+            python-version: pypy2
+          - os: windows-latest
+            python-version: pypy3
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Get full python version
-      id: full-python-version
-      run: |
-        echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
-    - name: Install and set up Poetry
-      run: |
-        curl -fsS -o get-poetry.py https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py
-        python get-poetry.py --preview -y
-        source $HOME/.poetry/env
-        poetry config virtualenvs.in-project true
-    - name: Set up cache
-      uses: actions/cache@v1
-      with:
-        path: .venv
-        key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
-    - name: Install dependencies
-      run: |
-        source $HOME/.poetry/env
-        poetry install
-    - name: Test
-      run: |
-        source $HOME/.poetry/env
-        poetry run pytest -q tests
-        poetry install
+      - uses: actions/checkout@v2
 
-  MacOS:
-    needs: Linting
-    runs-on: macos-latest
-    strategy:
-      matrix:
-        python-version: [2.7, 3.5, 3.6, 3.7, 3.8, pypy2, pypy3]
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
 
-    steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Get full python version
-      id: full-python-version
-      run: |
-        echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
-    - name: Install and set up Poetry
-      run: |
-        curl -fsS -o get-poetry.py https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py
-        python get-poetry.py --preview -y
-        source $HOME/.poetry/env
-        poetry config virtualenvs.in-project true
-    - name: Set up cache
-      uses: actions/cache@v1
-      with:
-        path: .venv
-        key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
-    - name: Install dependencies
-      run: |
-        source $HOME/.poetry/env
-        poetry install
-    - name: Test
-      run: |
-        source $HOME/.poetry/env
-        poetry run pytest -q tests
-  Windows:
-    needs: Linting
-    runs-on: windows-latest
-    strategy:
-      matrix:
-        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
+      - name: Install and configure poetry
+        run: |
+          pip install --pre poetry
```

Comment:
I prefer the usage of the custom installer since it's the official recommended way to install Poetry, so if there is anything wrong with we will see it.

Reply:
@sdispater I agree we should test it, but I feel that it should be tested in the poetry project using a separate workflow on develop or master. And since `pip install` is also expected to work, I think this is also valuable. I'd like to keep the workflow itself trivial. This also avoids any path related issues, dealing with them reliably across OSes can be fragile. Additionally, this also avoids us having to download the unnecessary python version dependencies. But these are minor advantages.

If you feel strongly about retaining the use of `get-poetry` I can add that in.